proto/tendermint/rpc/grpc/types.proto

```diff
@@ -23,10 +54,56 @@ message ResponseBroadcastTx {
   tendermint.abci.ResponseDeliverTx deliver_tx = 2;
 }
 
+message BlockByHashResponse {
+  tendermint.types.Part block_part = 1;
+  bool                  is_last    = 2;
+}
+
+message BlockByHeightResponse {
+  tendermint.types.Part block_part = 1;
+  bool                  is_last    = 2;
+}
+
+message BlockMetaByHashResponse {
+  tendermint.types.BlockMeta block_meta = 1;
+}
+
+message BlockMetaByHeightResponse {
+  tendermint.types.BlockMeta block_meta = 1;
+}
+
+message CommitResponse {
+  int64                    height                = 1;
+  int32                    round                 = 2;
+  tendermint.types.BlockID block_id              = 3 [(gogoproto.nullable) = false, (gogoproto.customname) = "BlockID"];
+  repeated tendermint.types.CommitSig signatures = 4 [(gogoproto.nullable) = false];
+}
+
+message ValidatorSetResponse {
+  repeated tendermint.types.Validator validators         = 1;
+  tendermint.types.Validator          proposer           = 2;
+  int64                               total_voting_power = 3;
+}
+
+message NewHeightEvent {
+  int64 height = 1;
+  bytes hash   = 2;
+}
+
 //----------------------------------------
 // Service Definition
 
 service BroadcastAPI {
   rpc Ping(RequestPing) returns (ResponsePing);
   rpc BroadcastTx(RequestBroadcastTx) returns (ResponseBroadcastTx);
 }
+
+service BlockAPI {
+  rpc BlockByHash(BlockByHashRequest) returns (stream BlockByHashResponse);
+  rpc BlockMetaByHash(BlockMetaByHashRequest) returns (BlockMetaByHashResponse);
+  rpc BlockByHeight(BlockByHeightRequest) returns (stream BlockByHeightResponse);
+  rpc BlockMetaByHeight(BlockMetaByHeightRequest) returns (BlockMetaByHeightResponse);
+  rpc Commit(CommitRequest) returns (CommitResponse);
+  rpc ValidatorSet(ValidatorSetRequest) returns (ValidatorSetResponse);
+  rpc SubscribeNewHeights(SubscribeNewHeightsRequest) returns (stream NewHeightEvent);
```

Comment:
Is the expected usage for the node to request each of `Commit/ValidatorSet/BlockMeta` separately on every new `NewHeightEvent`?

Reply:
yes

Reply:
Did you discuss making a single request to get `Commit/ValidatorSet/BlockMeta` in a single request instead of 3 separate ones?

Reply:
Yes, and we decided to provide the endpoints separately and celestia-node makes the necessary requests.

Reply:
My worry is that we are doing 3 requests instead of 1, when we don't need these types individually requested in practise, so its not well justified. You could do these requests concurrently, but that just moves complexity to the client.

Reply:
do you want an extra endpoint that returns the three of them? any idea how to call it?

Reply:
Let's not add it yet. Wanna hear @walldiss thoughts on this

Reply:
FYI I created the `SignedBlock` endpoint which was designed specifically for node as it contains the necessary information in a single request: https://github.com/celestiaorg/celestia-core/blob/2ffec2fc5563174c1fd51a58d6258ae09e238f94/rpc/core/blocks.go#L141

Reply:
thanks. If they want that on grpc, I can create the discussed endpoint. The reason why we cant stream SignedBlock is the need to re-implement the chunking for that type while we already having it for blocks and its gonna take some time to havee

Reply:
Aren't we streaming `NewHeightEvent` and fetch other types as per current iface?

Reply:
new heights are being streamed, but also the block is being chunked and streamed instead of being sent in full.

Reply:
Ok, I see now. We could extend both `BlockByHashResponse` and `BlockByHeightResponse` (also not sure why we need two identical types) with fields containing metadata that per contract will be filled on the first message in the stream, while all further will only contain blockparts until the last one.

Reply:
I think that's unnecessarily complicated compared to having separate clear self explaining API calls that can be used without any additional context

Reply:
Performance matters more than API readability in this case. We are doing this for ourselves only, so the ability to understand API without context doesn't seem like a valuable goal in this case.

By the way, with Shwap, an FN can sync 2MB blocks with only 2GB of RAM, while BN needed 12GB at minimum to function during sync. Ideally, we need BN to keep with FN with memory usage and sync speed.

Reply:
A new simple endpoint could also work for us, returning everything at once. However, the validator would need to load everything in memory first before serving instead of iteratively reading parts out of the disk.  It doesn't matter much on our receiving side, as we have to load everything in memory to recompute the square.

Reply:
I see, does making two requests:
- getBlockInfo
- getBlock
reduce the performance significantly than receiving all in the same call?

If I understand correctly, these will be done on the same connection.  So only an extra request will be the difference. does it really matter this much? if so, I can make an extra endpoint that does that for you

Reply:
This depends on the connection between the Bridge and the Validator. If GRPC runs over HTTP/3, there won't be any performance penalty for concurrent requests. If GRPC is not over HTTP/2, then the penalty will depend on the connection quality, e.g. being on the same host will have only a few milliseconds overhead per block.

Besides, it's a question of where to keep the complexity. One request is the simplest for the client, while several concurrent requests need a way to handle that. As consumers, we would prefer an easier option.

Reply:
@Wondertan 47695c32ea34fed0cb1718576658365b449bd695 contains that data also sent in the same response as the block parts